Loop/Managers/LoopAppManager.swift

```diff
@@ -400,27 +400,68 @@ class LoopAppManager: NSObject {
         return false
     }
     
-    func askUserToConfirmCrashIfNecessary() {
-        deviceDataManager.pluginManager.availableSupports.forEach { supportUI in
-            if supportUI.loopNeedsReset {
-                alertManager.presentConfirmCrashAlert() { [weak self] completion in
+    func askUserToConfirmLoopReset() {
+        if UserDefaults.appGroup?.userRequestedLoopReset == true {
+            alertManager.presentLoopResetConfirmationAlert(
+                confirmAction: { [weak self] completion in
                     guard let pumpManager = self?.deviceDataManager.pumpManager else {
-                        supportUI.resetLoop()
+                        self?.resetLoop()
                         completion()
                         return
                     }
                     
                     pumpManager.prepareForDeactivation() { [weak self] error in
                         guard let error = error else {
-                            supportUI.resetLoop()
+                            self?.resetLoop()
                             completion()
                             return
                         }
                         self?.alertManager.presentCouldNotResetLoopAlert(error: error)
                     }
+                },
+                cancelAction: {
+                    UserDefaults.appGroup?.userRequestedLoopReset = false
                 }
+            )
+        }
+    }
+    
+    private func resetLoop() {
+        deviceDataManager.pluginManager.availableSupports.enumerated().forEach { index, supportUI in
```

Comment:
I believe if you have 3 support plugins, the order of calls will be like this:
```
support1.loopWillReset()
support1.loopDidReset()
support2.loopWilReset()
support2.loopDidReset()
support3.loopWilReset()
resetLoopDocuments()
resetLoopUserDefaults()
support3.loopDidReset()

Reply:
Fixed and opened a new PR https://github.com/tidepool-org/Loop/pull/570